src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -0,0 +1,232 @@
+        }
+
+        return HearingWindow.builder()
+                .firstDateTimeMustBe(getFirstDateTimeMustBe())
+                .dateRangeStart(dateRangeStart)
+                .dateRangeEnd(dateRangeEnd)
+                .build();
+    }
+
+    public static LocalDateTime getFirstDateTimeMustBe() {
+        // TODO Adjournments - Find out how to use adjournCase data to work this out, possibly related variables:
+        //      adjournCaseNextHearingDateType, adjournCaseNextHearingDateOrPeriod, adjournCaseNextHearingDateOrTime,
+        //      adjournCaseNextHearingFirstAvailableDateAfterDate, adjournCaseNextHearingFirstAvailableDateAfterPeriod
+        // TODO Future Work - Manual Override
+        return null;
+    }
+
+    public static int getHearingDuration(SscsCaseData caseData) {
+        // TODO Adjournments - Check this is the correct logic for Adjournments
+        // TODO Future Work - Manual Override
+        // TODO Dependant on SSCS-10116 - Will use Session Category Reference Data
+
+        if (isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())
+                && caseData.getAdjournCaseNextHearingListingDuration().matches("\\d+")
+                && Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration()) > 0) {
+
+            if ("sessions".equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                return Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration()) * DURATION_SESSIONS_MULTIPLIER;
+            }
+            if ("hours".equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                // TODO Adjournments - check no other measurement than hours, sessions and null
+                return Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration()) * DURATION_HOURS_MULTIPLIER;
+            }
+        }
+        if (nonNull(caseData.getBenefitCode()) && nonNull(caseData.getIssueCode())) {
+            // TODO Dependant on SSCS-10116 - Will use Session Category Reference Data
+            return 45;
+        }
+        return DURATION_DEFAULT;
+    }
+
+    public static String getHearingPriority(SscsCaseData caseData) {
+        // urgentCase Should go to top of queue in LA - also consider case created date
+        // Flag to Lauren - how  can this be captured in HMC queue?
+        // If there's an adjournment - date shouldn't reset - should also go to top priority
+
+        String hearingPriorityType = NORMAL;
+
+        // TODO Adjournment - Check what should be used to check if there is adjournment
+        // TODO Dependant on SSCS-10273 - Needed for enum values and logic
+        if (isYes(caseData.getUrgentCase()) || isYes(caseData.getAdjournCaseCanCaseBeListedRightAway())) {
+            hearingPriorityType = HIGH;
+        }
+
+        return hearingPriorityType;
+    }
+
+    public static Number getNumberOfPhysicalAttendees() {
+        // TODO Implementation to be done by SSCS-10260
```

Comment:
I don;t think the ticket reference is right here. You're both depending on each other

Reply:
We need to review all the tickets referencing 260 and probably raise new tickets to cover them